File: src/main/java/com/canyoncompanion/canyon_api/dtos/responses/DescentResponseDTO.java
package com.canyoncompanion.canyon_api.dtos.responses;

import com.canyoncompanion.canyon_api.model.enums.AquaticCharacter;
import com.canyoncompanion.canyon_api.model.enums.Commitment;
import com.canyoncompanion.canyon_api.model.enums.VerticalCharacter;
import lombok.*;

import java.time.LocalDateTime;
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class DescentResponseDTO {

    private Long id;

    private String name;

    private String location;

    private String province;

    private VerticalCharacter verticalCharacter;

    private AquaticCharacter aquaticCharacter;

    private Commitment commitment;

    private String descriptionLink;

    private String comments;

    // =========================
    // RELACIÓN SIMPLIFICADA
    // =========================
    private Long userId;

    private String username;

    // =========================
    // AUDITORÍA
    // =========================
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: src/main/java/com/canyoncompanion/canyon_api/dtos/responses/DescentResponseDTO.kt
package com.canyoncompanion.canyon_api.dtos.responses

import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor
import java.time.LocalDateTime

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
class DescentResponseDTO {
    private val id: Long? = null

    private val name: String? = null

    private val location: String? = null

    private val province: String? = null

    private val verticalCharacter: String? = null

    private val aquaticCharacter: String? = null

    private val commitment: String? = null

    private val descriptionLink: String? = null

    private val comments: String? = null

    // =========================
    // RELACIÓN SIMPLIFICADA
    // =========================
    private val userId: Long? = null

    private val username: String? = null

    // =========================
    // AUDITORÍA
    // =========================
    private val createdAt: LocalDateTime? = null

    private val updatedAt: LocalDateTime? = null
}